frontend/javascripts/viewer/view/action_bar_view.tsx

```diff
@@ -160,31 +167,67 @@ function CreateAnnotationButton() {
       maybeMappingName = mappingInfo.mappingName;
     }
 
-    const annotation = await createExplorational(
-      dataset.id,
-      "hybrid",
-      false,
-      fallbackLayerName,
-      maybeMappingName,
+    createExplorational(dataset.id, "hybrid", false, fallbackLayerName, maybeMappingName).then(
+      (annotation) => {
+        history.push(`/annotations/${annotation.id}${location.hash}`);
+      },
```

Comment:
why no `await` anymore? in general, the "async" behavior is lost now. `continueWithLayer` should be `async` and its call should be awaited (both in `onClick` and `handleLayerSelected`). that way, the UI should automatically show a spinner if the user clicks on a button and the server request takes a bit.

Reply:
right, that makes sense. During my tests the spinner behaviour was as expected, even with throttling enabled. Apart from code logic, is there a better way to catch this?

Reply:
for me, the current dev instance (probably the old code?) doesn't show any spinners when using the "create annotation" button (either in the navbar or in the modal if it exists). I throttled the internet speed to DSL 200 which is ultra slow (one has to wait several seconds after clicking the button).
you say you did the same and you saw a spinner?

Reply:
Yes :sweat_smile: I tested it locally though, from the newly added modal where the layer is selected. And even now if checkout the commit before the review I can see the brain spinner.

Reply:
Oh I see, this makes so much sense!